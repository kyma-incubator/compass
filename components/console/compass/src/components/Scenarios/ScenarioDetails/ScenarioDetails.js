import React, { useState } from 'react';
import PropTypes from 'prop-types';

import ScenarioDetailsHeader from './ScenarioDetailsHeader/ScenarioDetailsHeader';
import ScenarioApplications from './ScenarioApplications/ScenarioApplications';
import ScenarioRuntimes from './ScenarioRuntimes/ScenarioRuntimes.container';

import ScenarioNameContext from './ScenarioNameContext';

ScenarioDetails.propTypes = {
  scenarioName: PropTypes.string.isRequired,
};

export default function ScenarioDetails({ scenarioName }) {
  const [applicationsCount, setApplicationsCount] = useState(0);

  return (
    <ScenarioNameContext.Provider value={scenarioName}>
      <ScenarioDetailsHeader applicationsCount={applicationsCount} />
      <ScenarioApplications updateApplicationsCount={setApplicationsCount} />
      <ScenarioRuntimes />
    </ScenarioNameContext.Provider>
  );
}
