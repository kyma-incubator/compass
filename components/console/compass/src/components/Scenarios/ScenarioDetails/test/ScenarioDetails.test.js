import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ScenarioDetails from './../ScenarioDetails';

describe('ScenarioDetailsHeader', () => {
  it('Renders with minimal props', () => {
    const component = shallow(<ScenarioDetails scenarioName={'scenario'} />);
    let tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
