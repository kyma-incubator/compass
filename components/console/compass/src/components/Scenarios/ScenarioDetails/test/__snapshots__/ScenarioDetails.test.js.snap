// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScenarioDetailsHeader Renders with minimal props 1`] = `
<ContextProvider
  value="scenario"
>
  <ScenarioDetailsHeader
    applicationsCount={0}
  />
  <ScenarioApplications
    updateApplicationsCount={[Function]}
  />
  <fromRenderProps(Apollo(Apollo(Apollo(Apollo(ScenarioRuntimes))))) />
</ContextProvider>
`;
